PolyFill.cs

```diff
@@ -0,0 +1,91 @@
+﻿// PolyFill.cs - Used to fill polygon
+// ------------------------------------------------------------------------------------------------
+
+using System.Collections.Concurrent;
+
+namespace GrayBMP;
+
+#region class PolyFill ----------------------------------------------------------------------------
+class PolyFill {
+   #region MultiThread Implementation -------------------------------
+   // Fill the polygon using Task. For this to work, Begin and End should be removed from
+   // DrawHorizontalLines method of GrayBMP
+   public void FillTask (GrayBMP bmp, int color) {
+      (mBmp, mColor) = (bmp, color);
+      mBmp.Begin ();
+      mQueue.Clear ();
+      for (int i = 0; i < bmp.Height; i++) mQueue.Enqueue (i);
+      Task[] tasks = new Task[Environment.ProcessorCount];
+      for (int i = 0; i < tasks.Length; i++)
+         tasks[i] = Task.Run (TaskProc);
+      Task.WaitAll (tasks);
+      mBmp.End ();

```

Comment:
What's the performance difference between single threaded and multi-threaded implementations? Can you benchmark and let me know.

Reply:
Single threaded
Leaf: 48ms
Leaf: 45ms
Leaf: 36ms
Leaf: 46ms
Leaf: 53ms
Leaf: 53ms
Leaf: 38ms
Leaf: 51ms
Leaf: 40ms
Leaf: 35ms
Leaf: 36ms
Multi threaded
Leaf: 8ms
Leaf: 8ms
Leaf: 27ms
Leaf: 26ms
Leaf: 17ms
Leaf: 36ms
Leaf: 15ms
Leaf: 19ms
Leaf: 17ms
Leaf: 18ms
Leaf: 14ms
Leaf: 12ms
Around 4 times difference in my PC